### PR TITLE
Adress filter/aimec bugs and add resDF

### PR DESCRIPTION
**Pull request by awirb**
* fix some bug for filtering function 
* add comments to stats plots 
* fix bug in aimec warning
* add resDF to be saved for com1DFA results that can be used for statistical analysis

### CODE REVIEW

File: avaframe/com1DFA/com1DFA.py [hunk 1]
```diff
@@ -1311,6 +1311,9 @@ def DFAIterate(cfg, particles, fields, dem, simHash=''):
     massEntrained = []
     massTotal = []
 
+    # setup a result fields info data frame to save max values of fields and avalanche front
+    resDF = setupResDF(resTypesLast, cfg['VISUALISATION'].getboolean('createRangeTimeDiagram'))
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:1315]: Would you mind renaming this to something like resultsDF resuDF ... (same for the functions)?
res is very much preloaded with 'resistance in the context of com1dfa...

File: avaframe/com1DFA/com1DFA.py [hunk 2]
```diff
@@ -1447,9 +1456,76 @@ def DFAIterate(cfg, particles, fields, dem, simHash=''):
...
+            resDict['max' + resT] = [0.0]
+    if cfgRangeTime:
+        resDict['rangeList'] = [0.0]
+    resDF = pd.DataFrame.from_dict(resDict)
+    resDF = resDF.set_index('timeStep')
+
+    return resDF
+
+
+def getMaxMeanValues(resDF, fields, timeStep, resTypes, rangeValue=''):
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:1495]: We already have a getMaxMeanValues function in aimecTools, could that be used?

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com1DFA/com1DFA.py (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:1495]: Actually the getMaxMeanValues function here just appends the max value of the fields to the dataFrame and if provided the rangeValue, the actual computation of the max value is just one line and so I think not useful to call the aimec function, however, the function here could be renamed to addMaxValuesToDF?

File: avaframe/com1DFA/com1DFA.py (hunk 1 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:1315]: renamed to resultsDF and setupResultSDF now

**Review by fso42 — APPROVED** [review]

**Pull request merged**